cmake/Modules/FindBLAS.cmake

```diff
@@ -290,6 +290,7 @@ int main() {
   ELSE (BLAS_USE_CBLAS_DOT)
     SET(BLAS_USE_CBLAS_DOT FALSE)
   ENDIF (BLAS_USE_CBLAS_DOT)
+  SET(CMAKE_REQUIRED_LIBRARIES)
```

Comment:
@yinghai 
split to #11975

Reply:
This is used to avoid corrupting the following-up compiling checks e.g. CHECK_C_SOURCE_COMPILES etc. "CMAKE_REQUIRED_LIBRARIES" is a global setting and may be used by other modules as well. So we'd better clean it up after being used each time.